src/core/geometry/qgsregularpolygon.h

```diff
@@ -0,0 +1,213 @@
+/***************************************************************************
+                         qgsregularpolygon.h
+                         --------------
+    begin                : May 2017
+    copyright            : (C) 2017 by Loîc Bartoletti
+    email                : lbartoletti at tuxfamily dot org
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSREGULARPOLYGON_H
+#define QGSREGULARPOLYGON_H
+
+#include <QString>
+
+#include "qgis_core.h"
+#include "qgspointv2.h"
+#include "qgspolygon.h"
+#include "qgslinestring.h"
+#include "qgscircle.h"
+#include "qgstriangle.h"
+
+enum ConstructionOption
```

Comment:
Can you move this inside the QgsRegularPolygon class, and rename the options to InscribedCircle and CircumscribedCircle to keep with the standard formatting.

Also needs some docs for these options/enum.